src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -1,346 +1,332 @@
-        };
-        this.setContentRef = this.setContentRef.bind(this);
-        this.anchorRef = React.createRef();
-    }
+const PopoverReportActionContextMenu = forwardRef((props,ref) => {
+  const reportIDRef = useRef('0');
+  const typeRef = useRef('');
+  const reportActionRef = useRef({});
+  const selectionRef = useRef('');
+  const reportActionDraftMessageRef = useRef('');
 
-    componentDidMount() {
-        this.dimensionsEventListener = Dimensions.addEventListener('change', this.measureContextMenuAnchorPosition);
-    }
+  const cursorRelativePosition = useRef({
+      horizontal: 0,
+      vertical: 0,
+  });
 
-    shouldComponentUpdate(nextProps, nextState) {
-        const previousLocale = lodashGet(this.props, 'preferredLocale', CONST.LOCALES.DEFAULT);
-        const nextLocale = lodashGet(nextProps, 'preferredLocale', CONST.LOCALES.DEFAULT);
-        return (
-            this.state.isPopoverVisible !== nextState.isPopoverVisible ||
-            this.state.popoverAnchorPosition !== nextState.popoverAnchorPosition ||
-            this.state.isDeleteCommentConfirmModalVisible !== nextState.isDeleteCommentConfirmModalVisible ||
-            previousLocale !== nextLocale
-        );
-    }
+  // The horizontal and vertical position (relative to the screen) where the popover will display.
+  const popoverAnchorPosition= useRef({
+      horizontal: 0,
+      vertical: 0,
+  });
 
-    componentWillUnmount() {
-        if (!this.dimensionsEventListener) {
-            return;
-        }
-        this.dimensionsEventListener.remove();
-    }
+  const [instanceID, setInstanceID] = useState('');
 
-    /**
-     * Get the Context menu anchor position
-     * We calculate the achor coordinates from measureInWindow async method
-     *
-     * @returns {Promise<Object>}
-     */
-    getContextMenuMeasuredLocation() {
-        return new Promise((resolve) => {
-            if (this.contextMenuAnchor) {
-                (this.contextMenuAnchor.current || this.contextMenuAnchor).measureInWindow((x, y) => resolve({x, y}));
-            } else {
-                resolve({x: 0, y: 0});
-            }
-        });
-    }
 
-    /**
-     * Whether Context Menu is active for the Report Action.
-     *
-     * @param {Number|String} actionID
-     * @return {Boolean}
-     */
-    isActiveReportAction(actionID) {
-        return Boolean(actionID) && this.state.reportAction.reportActionID === actionID;
-    }
+  const [isPopoverVisible, setIsPopoverVisible] = useState(false);
+  const [isDeleteCommentConfirmModalVisible, setIsDeleteCommentConfirmModalVisible] = useState(false);
+  const [shouldSetModalVisibilityForDeleteConfirmation, setShouldSetModalVisibilityForDeleteConfirmation] = useState(true);
+  const [isArchivedRoom, setIsArchivedRoom] = useState(false);
+  const [isChronosReport, setIsChronosReport] = useState(false);
+  const [isPinnedChat, setIsPinnedChat] = useState(false);
+  const [isUnreadChat, setIsUnreadChat] = useState(false);
```

Comment:
Since there are so many states, we could merge them all into one "options" or "config" state to avoid rerendering 4, 5 times in a row when calling setState

Reply:
@barttom Brought to my attention that in React 18 we have automatic batching for sequential setStates like this. So we shouldn't need what I suggested here - making this comment outdated!

FYI: https://react.dev/blog/2022/03/08/react-18-upgrade-guide#automatic-batching